core/src/test/java/org/elasticsearch/search/aggregations/metrics/cardinality/InternalCardinalityTests.java

```diff
@@ -38,13 +37,21 @@
     private static List<HyperLogLogPlusPlus> algos;
     private static int p;
 
-    @Before
-    public void setup() {
+    @After //we force @After to have it run before ESTestCase#after otherwise it fails
+    @Override
+    public void setUp() {
         algos = new ArrayList<>();
         p = randomIntBetween(HyperLogLogPlusPlus.MIN_PRECISION, HyperLogLogPlusPlus.MAX_PRECISION);
     }
 
     @Override
+    public void tearDown() {
```

Comment:
Does this need to call super.tearDown() as well as it overrides LuceneTestCase#tearDown()?